### PR TITLE
HAL refactoring - remove GET endpoint

### PR DESCRIPTION
**Pull request by marco-scoppetta**
- Removed deprecated GET endpoint in GraqlController
 - Removed a lot of old code used to create "generated-relation" nodes
-  Refactored code used to build explanation 
 - Updated Dashboard to use new endpoints paths

### CODE REVIEW

**Comment by marco-scoppetta** [comment]
!rtg -   SQLMigratorTest.whenSQLQueryContainsFunction_MigrationCanAccessResultOfFunction:160 » NoSuchElement

**Comment by burukuru** [comment]
!rtg

**Comment by burukuru** [comment]
retest this please

**Comment by burukuru** [comment]
!rtg

**Comment by marco-scoppetta** [comment]
!rtg - forever same error   SQLMigratorTest.whenSQLQueryContainsFunction_MigrationCanAccessResultOfFunction:160 » NoSuchElement

**Comment by marco-scoppetta** [comment]
!rtg

**Comment by burukuru** [comment]
@fppt @aelred I think this one just needs a review and merge if you're happy.
The pr-merge job keeps failing because of cassandra failures. It runs the same tests as "unit-test-pull-request-grakn-titan" so there's no point in waiting for both to pass.

I'm hoping to get rid of unit-test-pull-request-grakn-titan soon!

File: grakn-dashboard/src/js/HAL/HALParser.js [hunk 1]
```diff
@@ -20,6 +20,8 @@
 
 import * as API from '../util/HALTerms';
 import * as Utils from './APIUtils';
+import _ from 'underscore';
```
Comment (aelred) [inline grakn-dashboard/src/js/HAL/HALParser.js]: JS is a strange place

File: grakn-graql/src/main/java/ai/grakn/graql/internal/printer/HALPrinter.java [hunk 2]
```diff
@@ -36,9 +39,39 @@
...
+
+        return json;
+    }
+
+    public Json halString(Concept concept, boolean inferred) {
+        String json = HALBuilder.renderHALConceptData(concept, inferred, 0, keyspace, 0, limitEmbedded);
         return Json.read(json);
     }
 
+    // TODO: Add 'inferred' flag to concept
```
Comment (aelred) [inline grakn-graql/src/main/java/ai/grakn/graql/internal/printer/HALPrinter.java]: 🐘

File: grakn-graql/src/main/java/ai/grakn/graql/internal/printer/HALPrinter.java [hunk 3]
```diff
@@ -36,9 +39,39 @@
...
+    private boolean isInferred(Var key, Concept concept, Answer answer) {
+        if (key == null || answer.getExplanation().isEmpty()) return false;
+
+        //TO-DO add support for attributes
```
Comment (aelred) [inline grakn-graql/src/main/java/ai/grakn/graql/internal/printer/HALPrinter.java]: 🐘

File: grakn-dashboard/src/js/HAL/HALParser.js (hunk 1 above)
Comment (fppt) [reply grakn-dashboard/src/js/HAL/HALParser.js]: I used to use this library back in the day. Agreed.

File: grakn-graql/src/main/java/ai/grakn/graql/internal/printer/HALPrinter.java (hunk 2 above)
Comment (marco-scoppetta) [reply grakn-graql/src/main/java/ai/grakn/graql/internal/printer/HALPrinter.java]: This is a TODO for the future, as I think it requires some work from Filipe/Kasper and a nice long discussion

File: grakn-graql/src/main/java/ai/grakn/graql/internal/printer/HALPrinter.java (hunk 3 above)
Comment (marco-scoppetta) [reply grakn-graql/src/main/java/ai/grakn/graql/internal/printer/HALPrinter.java]: Also we might not need this one, not sure, I will discuss it with Kasper. But definitely the future

**Comment by marco-scoppetta** [comment]
I will make another PR to remove the materialise flag, this one already contains enough changes I think

**Pull request merged**